python/rmm/_cuda/stream.pxd

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from libc.stdint cimport uintptr_t
+from libcpp cimport bool
+
+from rmm._lib.cuda_stream_view cimport cuda_stream_view
+
+
+cdef class Stream:
+    cdef public:
+        uintptr_t _ptr
+        object _owner
+
+    cdef cuda_stream_view view(self) nogil except *
+    cpdef bool is_default(self) except *
+    cpdef void synchronize(self) except *
```

Comment:
Can both of these be `nogil`?

Reply:
Not if they're `cpdef` :(

Reply:
I could have sworn you can still mark cpdef functions as `nogil`